solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java

```diff
@@ -389,6 +390,7 @@ private String getHostCoreName(
     SolrQuery q = new SolrQuery("*:*");
     q.add("fl", "id,desc_s,_core_:[core]").add(OMIT_HEADER, TRUE);
     p.accept(q);
+    // TODO why is this building a string instead of using the proper SolrClient???
```

Comment:
SolrClient will not send the request to COORDINATOR , so we can't  test the behaviour

Reply:
I don't understand what you mean. I pushed up ff1f547eada49b494f5943ff2bb0e95ef3f4a671 which does exactly what I said using SolrClient.

Reply:
if you use `SolrClient` how do you ensure that the request is actually going to coordinator node?

Reply:
Unless I misunderstand how this works `try (SolrClient solrClient = new Http2SolrClient.Builder(qaNode).build()) {` - this builds a SolrClient pointing to the qanode.

Reply:
>this builds a SolrClient pointing to the qanode.

that is not what it does. `Http2SolrClient.Builder(qaNode).build()` builds a client with the `qaNode` as the "seed node". Thereafter, it queries the `qaNode` to get the collection topology and finds out where the request should be sent. Basically, it behaves the similar to the Zk based provider but using the REST APIs

Reply:
Are you sure you don't mean `CloudHttp2SolrClient`? I don't think that the `Http2SolrClient` should know anything about the cloud topology.

Reply:
Oh, my bad. You are right . I confused it with `CloudHttp2SolrClient`